src/validator/collator.rs

```diff
@@ -961,7 +1009,8 @@ impl ExecutionManager {
             AsyncMessage::Copyleft(_) => collator_data.copyleft_msgs.push(in_msg_opt.ok_or_else(|| error!("Can't unwrap `in_msg_opt`"))?),
             _ => ()
         }
-        collator_data.block_full |= !collator_data.block_limit_status.fits(ParamLimitIndex::Normal);
+        let pruned_count = collator_data.pruned_count();
```

Comment:
The same lines are repeated several times, consider addeng new method to CollatorData
```
fn limit_fits(...) {
   let pruned_count = self.pruned_count(); 
   self.block_limit_status.fits(...)
}

```

Reply:
resolved